src/main/java/sprouts/Vars.java

```diff
@@ -729,14 +797,52 @@ default Vars<T> addAll( Iterable<T> items ) {
     }
 
     /**
-     * Iterates over the supplied property list, and appends
-     * the values of the list to {@code this} list.
+     * Iterates over the supplied properties, and
+     * adds their items to this list in the form of new
+     * property instances.
+     * This method deliberately adds copies instead of the properties
+     * themselves due to the provided type being {@link Vals},
+     * which implies a read only addition.
+     * If you want to add the actual properties objects directly
+     * (including all the listeners associated with the properties),
+     * use {@link #addAll(Vars)}.
+     *
+     * @param vals The properties, whose items should be added to this list
+     *             in the form of new {@link Var} properties.
+     * @return {@code this} list of properties, to allow for method chaining.
+     * @throws NullPointerException if the supplied property list is null.
+     * @throws IllegalArgumentException if the list allows {@code null} and at least one
+     *                                  property does not allow {@code null}.
+     */
+    default Vars<T> addAll( Vals<T> vals ) {
+        Objects.requireNonNull(vals);
+        Vars<T> cloned = allowsNull() ? Vars.ofNullable(type()) : Vars.of(type());
+        for ( int i = 0; i < vals.size(); i++ ) {
+            Val<T> toBeAdded = vals.at(i);
+            if ( !this.allowsNull() && toBeAdded.allowsNull() )
+                throw new IllegalArgumentException("Null values are not allowed in this property list.");
+
+            cloned.add(
+                this.allowsNull() ?
+                    Var.ofNullable(toBeAdded.type(), toBeAdded.orElseNull()) :
+                    Var.of(toBeAdded.get())
+            );
+        }
+        return addAll(cloned);
+    }
```

Comment:
This is inconsistent with the other add `Val` methods.

Here we allow to add non-nullable `Vals` to a nullable `Vars` list.

Reply:
I totally agree. We should be consistent with the other add `Val` methods.
And I would also prefer the flexible approach you describe.